src/software/ai/motion_constraint/motion_constraint_set_builder.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <set>
+
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/motion_constraint/motion_constraint.h"
+#include "software/ai/motion_constraint/motion_constraint_visitor.h"
+#include "software/world/game_state.h"
+
+class MotionConstraintSetBuilder
+{
+   public:
+    explicit MotionConstraintSetBuilder() = default;
```

Comment:
Can all these functions be static, i.e. delete this class entirely?

Reply:
Hmm, I think we can make everything static, so would we remove the `current_allowed_constraints` variable, and in MotionConstraintSetBuilder, we just create a set, and return it? Also, what do you mean by deleting the class entirely?

Reply:
Essentially, we don't need a `MotionConstraintSetBuilder` class at all. `motion_constraint_set_builder.h` can define some functions that can be used directly without first constructing a class